amm-executor/src/Spectrum/Executor/OrdersExecutor/Process.hs

```diff
@@ -67,27 +69,29 @@ mkOrdersExecutor
     , HasType (MakeLogging f m) env
     )
   => BacklogService m
+  -> QSem
   -> Transactions m era
   -> PoolResolver m
   -> PoolActions
   -> f (OrdersExecutor s m)
-mkOrdersExecutor backlog transactions resolver poolActions = do
+mkOrdersExecutor backlog syncSem transactions resolver poolActions = do
   MakeLogging{..} <- askContext
   logging         <- forComponent "OrdersExecutor"
   pure $ OrdersExecutor
-    { run = run' logging backlog transactions resolver poolActions
+    { run = run' logging syncSem backlog transactions resolver poolActions
     }
 
 run'
   :: forall s m era. (IsStream s, MonadAsync m, MonadUnliftIO m)
   => Logging m
+  -> QSem
   -> BacklogService m
   -> Transactions m era
   -> PoolResolver m
   -> PoolActions
   -> s m ()
-run' logging@Logging{..} backlog@BacklogService{..} txs resolver poolActions =
-  S.repeatM tryAcquire & S.mapM (\case
+run' logging@Logging{..} syncSem backlog@BacklogService{..} txs resolver poolActions =
+  S.before (liftIO $ waitQSem syncSem) $ S.repeatM tryAcquire & S.mapM (\case
```

Comment:
Why is Sem needed here?

Reply:
There is no necessary to start processing orders during sync process